File: Codeforces/div731/A.cpp

#include <iostream>

using namespace::std;

bool inside(int __x1, int __x2, int a) {
	int x1 = min(__x1, __x2);
	int x2 = max(__x1, __x2);

	if (x1 < a and a < x2) {
		return (true);
	}
	return (false);
}

int solve(int ax, int ay, int bx, int by, int fx, int fy) {
	 if ((ax == bx and bx == fx and inside(ay, by, fy)) or (ay == by and by == fy and inside(ax, bx, fx))) {
	 	return (abs(ax - bx) + abs(ay - by) + 2);
	 }
	 else {
	 	return (abs(ax - bx) + abs(ay - by));
	 }
}

int main() {
	int t;
	char nl;
	int ax, ay, bx, by, fx, fy;

	cin >> t;
	while (t--) {
		cin >> ax >> ay >> bx >> by >> fx >> fy;
		cout << solve(ax, ay, bx, by, fx, fy) << endl;
	}
}

File: Codeforces/div731/A.py
def inside(x1, x2, a):
	x1, x2 = min(x1, x2), max(x1, x2)
	if x1 < a < x2:
		return True
	return False


for i in range(int(input())):
	input()
	ax, ay = map(int, input().split())
	bx, by = map(int, input().split())
	fx, fy = map(int, input().split())

	if (ax == bx == fx and inside(ay, by, fy)) or (ay == by == fy and inside(ax, bx, fx)):
		print(abs(ax - bx) + abs(ay - by) + 2)
	else:
		print(abs(ax - bx) + abs(ay - by))
